### DISCUSSION TITLE
Read Input Register failed

### DISCUSSION THREAD

**Discussion by Uwdmic** [camueller/SmartApplianceEnabler#328] [Installation und Betrieb]
![A8943441-1A2C-4F68-9680-5F07D7BF563D](https://user-images.githubusercontent.com/112196531/187997639-935a02c0-43c5-48b1-aa1f-685ee2278858.jpeg)
![62947549-1218-4AC5-8F6E-49C8D41541D8](https://user-images.githubusercontent.com/112196531/187997829-c30e192a-aeca-4f44-abb0-fd3ba387db8d.jpeg)
![59DC7042-61D0-4BEF-85EA-4C5E49BD778C](https://user-images.githubusercontent.com/112196531/187997869-a21a9554-d5fd-48f6-8712-545bce4a6e5f.jpeg)
![09702C13-1E4B-44D9-A0E8-C877B8676862](https://user-images.githubusercontent.com/112196531/187997881-33f6f3ff-9abd-4afc-ab4c-d8cc7d0ee5ed.jpeg)

SDM72D-M mit ca 30cm Kabel und zwei Widerständen , wie in der Doku beschrieben, an USB Adapter , im Raspy

Gerät wird in SunnyPortal als E-Herd erkannt, aber es werden keine Werte angezeigt

Am SDM72 sollte lt Betriebsanleitung ein Telefonzeichen leuchten, als Anzeige für die Kommunikation, tut es aber nicht

Wo bitte könnte mein Fehler liegen

Danke 
LG
Michael

**Comment by camueller** [comment]
Solange Du mit `mbpoll` oder einem anderen Modbus-Client nicht erfolgreich den Zähler auslesen kannst, brauchst Du es mit dem SAE gar nicht erst versuchen und auch das Sunny Portal ist erstmal egal. Ich weiss nicht, was für einen Adapter Du verwendest, aber Modbus/RTU ist ein 2-Draht-Bus, aber an Deinem Adapter sind 3 Drähte.

Dein vorletzter Screenshot zeigt ausserdem, dass Du den Zähler als Gerät angelegt hast. Bitte lies Dir die Dokumentation nochmal durch - Du hast den grundsätzlichen Zusammenhang zwischen Gerät (Appliance), Zähler und Schalter nicht verstanden.

Auch  die von Dir verwendete ID wird wohl eher nicht funktionieren. Auch dazu steht einiges in der Dokumentation.

**Comment by Uwdmic** [comment]
Danke erstmal

Ich verwende diesen Adapter - DSD TECH USB zu RS485 Konverter SH-U10 [Adapter mit CP2102 Chip Kompatibel mit Windows 7,8,10, Linux, Mac OS](https://www.amazon.de/gp/product/B078X5H8H7/ref=ppx_yo_dt_b_asin_title_o00_s00?ie=UTF8&psc=1)
kannst du mir bitte einen Adapter empfehlen, falls der nicht passen sollte 

das 3.Kabel war GND, hab ich jetzt enfernt, 
--- Polling slave 1 ---
[342]:  0,085
schaut mal gut aus

Edit: 15 min später
--- Polling slave 1 ---
[342]:  0,272

bzgl Gerät - Zähler, ist das so nicht richtig ? - im Portal kommen noch keine Werte an

![SAE](https://user-images.githubusercontent.com/112196531/188126963-435bd15d-0710-4a7d-a7f5-95a9abbfe922.jpg)

**Comment by camueller** [comment]
Siehe https://github.com/camueller/SmartApplianceEnabler/blob/master/doc/QA_DE.md#semp2
und https://github.com/camueller/SmartApplianceEnabler/blob/master/doc/ModbusMeter_DE.md#log

**Comment by Uwdmic** [comment]
> mbpoll -a 1 -r 342 -t 3:float -B -0 -1 localhost
mbpoll 1.4-12 - FieldTalk(tm) Modbus(R) Master Simulator
Copyright © 2015-2019 Pascal JEAN, https://github.com/epsilonrt/mbpoll
This program comes with ABSOLUTELY NO WARRANTY.
This is free software, and you are welcome to redistribute it
under certain conditions; type 'mbpoll -w' for details.
Protocol configuration: Modbus TCP
Slave configuration...: address = [1]
                        start reference = 342, count = 1
Communication.........: localhost, port 502, t/o 1.00 s, poll rate 1000 ms
Data type.............: 32-bit float (big endian), input register table
-- Polling slave 1...
[342]:  0.272

> 2022-09-02 23:59:10,768 ERROR [Timer-0] d.a.s.m.ModbusElectricityMeter [ModbusElectricityMeter.java:236] F-0001-000000000010-00: Error reading input register 0x0156

ich hab gestern mehrere Std mit einem Freund gemeinsam die Doku durchgearbeitet, leider sind wir nicht weiter gekommen

![XMLfile](https://user-images.githubusercontent.com/112196531/188268788-dc23c4c8-aa13-403a-81ab-df63aa236095.jpg)

**Comment by camueller** [comment]
Falls Du nicht weiterkommst, kann ich Dir [kostenpflichtigen Support](https://github.com/camueller/SmartApplianceEnabler/blob/master/doc/Support_DE.md) anbieten.

**Reply by Uwdmic** [reply]
Danke

gerne nehme ich dein Angebot an, der Zugriff wäre bereits möglich

**Reply by camueller** [reply]
Passt Dir morgen (Montag) abend ab 18 Uhr?

**Reply by Uwdmic** [reply]
Sorry, MO geht nicht, da ich ab 18:00 (Nachtschicht) arbeite, aber gerne Dienstag, da kann ich später anfangen
